projectile-rails.el

```diff
@@ -970,28 +970,36 @@ The bound variable is \"filename\"."
 
 (defun projectile-rails-find-constant (name)
   (let* ((code-dirs (-filter #'f-exists? (-map #'projectile-rails-expand-root (projectile-rails--code-directories))))
-         (file-name (format "%s.rb" (projectile-rails-declassify name)))
          (list-parent-dirs (lambda (some-file)
                              (let ((parent-dirs '()))
                                (f-traverse-upwards (lambda (parent-dir)
                                                      (push (f-canonical parent-dir) parent-dirs)
                                                      (equal (f-slash (f-canonical (projectile-rails-root))) (f-slash (f-canonical parent-dir))))
                                                    (f-dirname some-file))
                                parent-dirs)))
-         (lookup-dirs (-flatten (list
-                                 ;; Look in current file namespace
-                                 (f-no-ext buffer-file-name)
-                                 ;; Look in local namespace hierarchy
-                                 (funcall list-parent-dirs buffer-file-name)
-                                 ;; Look in code directories
-                                 code-dirs)))
-         (lookup-paths (--map (f-join it file-name)
+         (file-name (format "%s.rb" (projectile-rails-declassify name)))
+         (lookup-dirs (if (f-absolute? file-name)
+                          ;; If top-level constant (e.g. ::Classname), i.e. derived filename (/classname) starts with a "/", then:
+                          ;; Look only in code directories
+                          code-dirs
+                        (-flatten (list
+                                   ;; Otherwise (relative constant):
+                                   ;; 1. Look in current file namespace
+                                   (f-no-ext buffer-file-name)
+                                   ;; 2. Look in local namespace hierarchy
+                                   (funcall list-parent-dirs buffer-file-name)
+                                   ;; 3. Look in code directories
+                                   code-dirs))))
+         ;; Strip leading "/" if present before generating lookup paths because it messes with f-join)
+         (relative-file-name (if (f-absolute? file-name)
+                                 (substring file-name 1)
+                               file-name))
+         (lookup-paths (--map (f-join it relative-file-name)
```

Comment:
@ejoubaud maybe we could get away with the `if` above by refactoring the `--map` like that:
```el
(--map (f-join it (replace-regexp-in-string "^/?\\(.+\\)" "\\1" relative-file-name)))
```
It's just I don't like "ifs" :) . Not sure about it but maybe the elisp virtual machine will be smart enough to optimize this call so there will be no performance penalty. 
Let me know what you think.